allennlp/data/dataset_readers/coreference_resolution/conll.py

```diff
@@ -0,0 +1,197 @@
+import logging
+import re
+import collections
+from typing import Any, Dict, List, Optional, Tuple
+
+from overrides import overrides
+
+from allennlp.common import Params
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.file_utils import cached_path
+from allennlp.data.dataset import Dataset
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.fields import Field, ListField, TextField, IndexField, MetadataField, SequenceLabelField
+from allennlp.data.instance import Instance
+from allennlp.data.tokenizers import Token
+from allennlp.data.token_indexers import SingleIdTokenIndexer, TokenIndexer
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+@DatasetReader.register("coref")
+class ConllCorefReader(DatasetReader):
+    def __init__(self,
+                 max_span_width: int,
+                 token_indexers: Dict[str, TokenIndexer] = None) -> None:
+        self._max_span_width = max_span_width
+        self._token_indexers = token_indexers or {"tokens": SingleIdTokenIndexer()}
+        self._begin_doc_regex = re.compile(r"#begin document \((.*)\); part (\d+)")
+
+    @overrides
+    def read(self, file_path: str):
+        # if `file_path` is a URL, redirect to the cache
+        file_path = cached_path(file_path)
+
+        logger.info("Reading file at %s", file_path)
+        instances = []
+        with open(file_path) as dataset_file:
+            document_state = DocumentState()
+            for line in dataset_file.readlines():
+                if self.handle_line(line, document_state):
+                    clusters = document_state.finalize()
+                    instance = self.text_to_instance(document_state.sentences, clusters)
+                    instances.append(instance)
+                    document_state = DocumentState()
+        if not instances:
+            raise ConfigurationError("No instances were read from the given filepath {}. "
+                                     "Is the path correct?".format(file_path))
+        return Dataset(instances)
+
+    @overrides
+    def text_to_instance(self,  # type: ignore
+                         sentences: List[List[str]],
+                         clusters: List[List[Tuple[int, int]]] = None) -> Instance:
```

Comment:
This is a complex enough type that a short docstring would be useful, e.g are the spans inclusive/exclusive.

Reply:
Added docstring.